Adopt more smart pointers in WebProcessPool

cdumez opened the pull request:
#### 006bf284d44b8c07fa0a6a74879db09925878b11
<pre>
Adopt more smart pointers in WebProcessPool
<a href="https://bugs.webkit.org/show_bug.cgi?id=263839">https://bugs.webkit.org/show_bug.cgi?id=263839</a>

Reviewed by Brent Fulgham.

* Source/WebKit/Shared/WebPreferencesStore.h:
* Source/WebKit/UIProcess/API/APIPageConfiguration.cpp:
(API::PageConfiguration::setProcessPool):
(API::PageConfiguration::setUserContentController):
(API::PageConfiguration::setWebExtensionController):
(API::PageConfiguration::setPageGroup):
(API::PageConfiguration::setPreferences):
(API::PageConfiguration::setRelatedPage):
(API::PageConfiguration::setVisitedLinkStore):
(API::PageConfiguration::protectedWebsiteDataStore):
(API::PageConfiguration::setWebsiteDataStore):
(API::PageConfiguration::setDefaultWebsitePolicies):
(API::PageConfiguration::setApplicationManifest):
* Source/WebKit/UIProcess/API/APIPageConfiguration.h:
* Source/WebKit/UIProcess/SuspendedPageProxy.h:
* Source/WebKit/UIProcess/WebPageGroup.cpp:
(WebKit::WebPageGroup::protectedPreferences const):
(WebKit::WebPageGroup::protectedUserContentController):
* Source/WebKit/UIProcess/WebPageGroup.h:
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::preferencesStore const):
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/WebProcessPool.cpp:
(WebKit::processPools):
(WebKit::WebProcessPool::allProcessPools):
(WebKit::WebProcessPool::~WebProcessPool):
(WebKit::WebProcessPool::setOverrideLanguages):
(WebKit::WebProcessPool::sendMemoryPressureEvent):
(WebKit::WebProcessPool::setApplicationIsActive):
(WebKit::WebProcessPool::networkProcessDidTerminate):
(WebKit::WebProcessPool::ensureGPUProcess):
(WebKit::WebProcessPool::ensureProtectedGPUProcess):
(WebKit::WebProcessPool::gpuProcessDidFinishLaunching):
(WebKit::WebProcessPool::gpuProcessExited):
(WebKit::WebProcessPool::createGPUProcessConnection):
(WebKit::WebProcessPool::establishRemoteWorkerContextConnectionToNetworkProcess):
(WebKit::WebProcessPool::windowServerConnectionStateChanged):
(WebKit::WebProcessPool::tryTakePrewarmedProcess):
(WebKit::WebProcessPool::initializeNewWebProcess):
(WebKit::WebProcessPool::enableProcessTermination):
(WebKit::WebProcessPool::processDidFinishLaunching):
(WebKit::WebProcessPool::disconnectProcess):
(WebKit::WebProcessPool::processForRegistrableDomain):
(WebKit::WebProcessPool::createWebPage):
(WebKit::WebProcessPool::updateRemoteWorkerUserAgent):
(WebKit::WebProcessPool::pageEndUsingWebsiteDataStore):
(WebKit::WebProcessPool::download):
(WebKit::WebProcessPool::resumeDownload):
(WebKit::WebProcessPool::postMessageToInjectedBundle):
(WebKit::WebProcessPool::handleMemoryPressureWarning):
(WebKit::WebProcessPool::activePagesOriginsInWebProcessForTesting):
(WebKit::WebProcessPool::registerURLSchemeAsSecure):
(WebKit::WebProcessPool::registerURLSchemeAsBypassingContentSecurityPolicy):
(WebKit::WebProcessPool::registerURLSchemeAsLocal):
(WebKit::WebProcessPool::registerURLSchemeAsNoAccess):
(WebKit::WebProcessPool::registerGlobalURLSchemeAsHavingCustomProtocolHandlers):
(WebKit::WebProcessPool::unregisterGlobalURLSchemeAsHavingCustomProtocolHandlers):
(WebKit::WebProcessPool::updateBackForwardCacheCapacity):
(WebKit::WebProcessPool::setCacheModel):
(WebKit::WebProcessPool::setCacheModelSynchronouslyForTesting):
(WebKit::WebProcessPool::createDownloadProxy):
(WebKit::WebProcessPool::terminateAllWebContentProcesses):
(WebKit::WebProcessPool::terminateServiceWorkers):
(WebKit::WebProcessPool::setAutomationSession):
(WebKit::webProcessProxyFromConnection):
(WebKit::WebProcessPool::handleMessage):
(WebKit::WebProcessPool::handleSynchronousMessage):
(WebKit::WebProcessPool::startedUsingGamepads):
(WebKit::WebProcessPool::stoppedUsingGamepads):
(WebKit::WebProcessPool::gamepadConnected):
(WebKit::WebProcessPool::gamepadDisconnected):
(WebKit::WebProcessPool::updateProcessAssertions):
(WebKit::WebProcessPool::processForNavigation):
(WebKit::WebProcessPool::processForNavigationInternal):
(WebKit::WebProcessPool::addMockMediaDevice):
(WebKit::WebProcessPool::clearMockMediaDevices):
(WebKit::WebProcessPool::removeMockMediaDevice):
(WebKit::WebProcessPool::setMockMediaDeviceIsEphemeral):
(WebKit::WebProcessPool::resetMockMediaDevices):
(WebKit::WebProcessPool::setDomainsWithCrossPageStorageAccess):
(WebKit::WebProcessPool::seedResourceLoadStatisticsForTesting):
(WebKit::WebProcessPool::notifyMediaStreamingActivity):
(WebKit::WebProcessPool::setUseSeparateServiceWorkerProcess):
(WebKit::WebProcessPool::forEachProcessForSession):
* Source/WebKit/UIProcess/WebProcessPool.h:
* Source/WebKit/UIProcess/WebProcessProxy.h:
(WebKit::WebProcessProxy::protectedWebConnection const):
* Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.cpp:
(WebKit::allDataStores):
(WebKit::WebsiteDataStore::forEachWebsiteDataStore):
(WebKit::WebsiteDataStore::existingDataStoreForIdentifier):
(WebKit::WebsiteDataStore::dataStoreForIdentifier):
(WebKit::WebsiteDataStore::registerWithSessionIDMap):
(WebKit::WebsiteDataStore::forwardAppBoundDomainsToITPIfInitialized):
(WebKit::WebsiteDataStore::forwardManagedDomainsToITPIfInitialized):

Canonical link: <a href="https://commits.webkit.org/269971@main">https://commits.webkit.org/269971@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c8cd2534d21365a4fb0deb91ee31633133162d6e

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/24009 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/2120 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/25097 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/26150 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/22115 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/3740 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/24492 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/22605 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/24253 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/1650 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/20750 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/26739 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/1406 "Passed tests") | | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/27901 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/21883 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/21953 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/25690 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/1346 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/19021 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/1364 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/5786 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/1756 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/1697 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

brentfulgham commented on Source/WebKit/UIProcess/WebProcessPool.cpp:
```diff
@@ -510,39 +515,38 @@ void WebProcessPool::createGPUProcessConnection(WebProcessProxy& webProcessProxy
-    WebProcessPool* processPool = requestingProcess ? &requestingProcess->processPool() : processPools()[0];
-    ASSERT(processPool);
+    RefPtr requestingProcess = requestingProcessIdentifier ? WebProcessProxy::processForIdentifier(*requestingProcessIdentifier) : nullptr;
+    Ref processPool = requestingProcess ? requestingProcess->processPool() : processPools()[0].get();
```
Nit: Maybe ` : processPools().first().get();`

brentfulgham commented on Source/WebKit/UIProcess/WebProcessPool.cpp:
```diff
@@ -617,9 +621,8 @@ void WebProcessPool::removeRemoteWorkerProcess(WebProcessProxy& process)
-    size_t processCount = m_processes.size();
-    for (size_t i = 0; i < processCount; ++i)
-        m_processes[i]->windowServerConnectionStateChanged();
+    for (Ref process : m_processes)
```
Wow! We missed an opportunity to use modern C++ loops at some point.

brentfulgham reviewed (approved):
LGTM. r=me

cdumez committed:
Adopt more smart pointers in WebProcessPool
https://bugs.webkit.org/show_bug.cgi?id=263839

Reviewed by Brent Fulgham.

* Source/WebKit/Shared/WebPreferencesStore.h:
* Source/WebKit/UIProcess/API/APIPageConfiguration.cpp:
(API::PageConfiguration::setProcessPool):
(API::PageConfiguration::setUserContentController):
(API::PageConfiguration::setWebExtensionController):
(API::PageConfiguration::setPageGroup):
(API::PageConfiguration::setPreferences):
(API::PageConfiguration::setRelatedPage):
(API::PageConfiguration::setVisitedLinkStore):
(API::PageConfiguration::protectedWebsiteDataStore):
(API::PageConfiguration::setWebsiteDataStore):
(API::PageConfiguration::setDefaultWebsitePolicies):
(API::PageConfiguration::setApplicationManifest):
* Source/WebKit/UIProcess/API/APIPageConfiguration.h:
* Source/WebKit/UIProcess/SuspendedPageProxy.h:
* Source/WebKit/UIProcess/WebPageGroup.cpp:
(WebKit::WebPageGroup::protectedPreferences const):
(WebKit::WebPageGroup::protectedUserContentController):
* Source/WebKit/UIProcess/WebPageGroup.h:
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::preferencesStore const):
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/WebProcessPool.cpp:
(WebKit::processPools):
(WebKit::WebProcessPool::allProcessPools):
(WebKit::WebProcessPool::~WebProcessPool):
(WebKit::WebProcessPool::setOverrideLanguages):
(WebKit::WebProcessPool::sendMemoryPressureEvent):
(WebKit::WebProcessPool::setApplicationIsActive):
(WebKit::WebProcessPool::networkProcessDidTerminate):
(WebKit::WebProcessPool::ensureGPUProcess):
(WebKit::WebProcessPool::ensureProtectedGPUProcess):
(WebKit::WebProcessPool::gpuProcessDidFinishLaunching):
(WebKit::WebProcessPool::gpuProcessExited):
(WebKit::WebProcessPool::createGPUProcessConnection):
(WebKit::WebProcessPool::establishRemoteWorkerContextConnectionToNetworkProcess):
(WebKit::WebProcessPool::windowServerConnectionStateChanged):
(WebKit::WebProcessPool::tryTakePrewarmedProcess):
(WebKit::WebProcessPool::initializeNewWebProcess):
(WebKit::WebProcessPool::enableProcessTermination):
(WebKit::WebProcessPool::processDidFinishLaunching):
(WebKit::WebProcessPool::disconnectProcess):
(WebKit::WebProcessPool::processForRegistrableDomain):
(WebKit::WebProcessPool::createWebPage):
(WebKit::WebProcessPool::updateRemoteWorkerUserAgent):
(WebKit::WebProcessPool::pageEndUsingWebsiteDataStore):
(WebKit::WebProcessPool::download):
(WebKit::WebProcessPool::resumeDownload):
(WebKit::WebProcessPool::postMessageToInjectedBundle):
(WebKit::WebProcessPool::handleMemoryPressureWarning):
(WebKit::WebProcessPool::activePagesOriginsInWebProcessForTesting):
(WebKit::WebProcessPool::registerURLSchemeAsSecure):
(WebKit::WebProcessPool::registerURLSchemeAsBypassingContentSecurityPolicy):
(WebKit::WebProcessPool::registerURLSchemeAsLocal):
(WebKit::WebProcessPool::registerURLSchemeAsNoAccess):
(WebKit::WebProcessPool::registerGlobalURLSchemeAsHavingCustomProtocolHandlers):
(WebKit::WebProcessPool::unregisterGlobalURLSchemeAsHavingCustomProtocolHandlers):
(WebKit::WebProcessPool::updateBackForwardCacheCapacity):
(WebKit::WebProcessPool::setCacheModel):
(WebKit::WebProcessPool::setCacheModelSynchronouslyForTesting):
(WebKit::WebProcessPool::createDownloadProxy):
(WebKit::WebProcessPool::terminateAllWebContentProcesses):
(WebKit::WebProcessPool::terminateServiceWorkers):
(WebKit::WebProcessPool::setAutomationSession):
(WebKit::webProcessProxyFromConnection):
(WebKit::WebProcessPool::handleMessage):
(WebKit::WebProcessPool::handleSynchronousMessage):
(WebKit::WebProcessPool::startedUsingGamepads):
(WebKit::WebProcessPool::stoppedUsingGamepads):
(WebKit::WebProcessPool::gamepadConnected):
(WebKit::WebProcessPool::gamepadDisconnected):
(WebKit::WebProcessPool::updateProcessAssertions):
(WebKit::WebProcessPool::processForNavigation):
(WebKit::WebProcessPool::processForNavigationInternal):
(WebKit::WebProcessPool::addMockMediaDevice):
(WebKit::WebProcessPool::clearMockMediaDevices):
(WebKit::WebProcessPool::removeMockMediaDevice):
(WebKit::WebProcessPool::setMockMediaDeviceIsEphemeral):
(WebKit::WebProcessPool::resetMockMediaDevices):
(WebKit::WebProcessPool::setDomainsWithCrossPageStorageAccess):
(WebKit::WebProcessPool::seedResourceLoadStatisticsForTesting):
(WebKit::WebProcessPool::notifyMediaStreamingActivity):
(WebKit::WebProcessPool::setUseSeparateServiceWorkerProcess):
(WebKit::WebProcessPool::forEachProcessForSession):
* Source/WebKit/UIProcess/WebProcessPool.h:
* Source/WebKit/UIProcess/WebProcessProxy.h:
(WebKit::WebProcessProxy::protectedWebConnection const):
* Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.cpp:
(WebKit::allDataStores):
(WebKit::WebsiteDataStore::forEachWebsiteDataStore):
(WebKit::WebsiteDataStore::existingDataStoreForIdentifier):
(WebKit::WebsiteDataStore::dataStoreForIdentifier):
(WebKit::WebsiteDataStore::registerWithSessionIDMap):
(WebKit::WebsiteDataStore::forwardAppBoundDomainsToITPIfInitialized):
(WebKit::WebsiteDataStore::forwardManagedDomainsToITPIfInitialized):

Canonical link: https://commits.webkit.org/269971@main
